openssl-dynamic/src/main/c/sslcontext.c

```diff
@@ -1652,3 +1653,121 @@ TCN_IMPLEMENT_CALL(void, SSLContext, disableOcsp)(TCN_STDARGS, jlong ctx) {
+  { TCN_METHOD_TABLE_ENTRY(setSessionCacheSize, (JJ)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(getSessionCacheSize, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionNumber, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionConnect, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionConnectGood, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionConnectRenegotiate, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionAccept, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionAcceptGood, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionAcceptRenegotiate, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionHits, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionCbHits, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionMisses, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionTimeouts, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionCacheFull, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionTicketKeyNew, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionTicketKeyResume, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionTicketKeyRenew, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(sessionTicketKeyFail, (J)J, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(setSessionTicketKeys0, (J[B)V, SSLContext) },
+
+  // setCertVerifyCallback -> needs dynamic method table
+  // setCertRequestedCallback -> needs dynamic method table
+  // setSniHostnameMatcher -> needs dynamic method table
+
+  { TCN_METHOD_TABLE_ENTRY(setSessionIdContext, (J[B)Z, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(setMode, (JI)I, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(getMode, (J)I, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(enableOcsp, (JZ)V, SSLContext) },
+  { TCN_METHOD_TABLE_ENTRY(disableOcsp, (J)V, SSLContext) }
+};
+
+static const jint fixed_method_table_size = sizeof(fixed_method_table) / sizeof(fixed_method_table[0]);
+
+static jint dynamicMethodsTableSize() {
+    return fixed_method_table_size + 3;
+}
+
+static JNINativeMethod* createDynamicMethodsTable(const char* packagePrefix) {
+    JNINativeMethod* dynamicMethods = malloc(sizeof(JNINativeMethod) * dynamicMethodsTableSize());
+    memcpy(dynamicMethods, fixed_method_table, sizeof(fixed_method_table));
+    char* dynamicTypeName = netty_internal_tcnative_util_prepend(packagePrefix, "io/netty/internal/tcnative/CertificateVerifier;");
+    char* tmpMethodSignature = netty_internal_tcnative_util_prepend("(JL", dynamicTypeName);
+    JNINativeMethod* dynamicMethod = &dynamicMethods[fixed_method_table_size];
+    dynamicMethod->name = "setCertVerifyCallback";
+    dynamicMethod->signature = netty_internal_tcnative_util_prepend(tmpMethodSignature, ")V");
+    dynamicMethod->fnPtr = (void *) TCN_FUNCTION_NAME(SSLContext, setCertVerifyCallback);
+    free(dynamicTypeName);
+    free(tmpMethodSignature);
+
+    dynamicTypeName = netty_internal_tcnative_util_prepend(packagePrefix, "io/netty/internal/tcnative/CertificateRequestedCallback;");
```

Comment:
`io/netty/internal/tcnative/CertificateRequestedCallback;` -> `io/netty/internal/tcnative/CertificateRequestedCallback;)V`. Then you can get rid of the temporary allocations/copies (`netty_internal_tcnative_util_prepend(tmpMethodSignature, ")V")`) and not have to worry about freeing more memory etc...